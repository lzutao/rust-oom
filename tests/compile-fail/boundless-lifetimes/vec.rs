use oom::{NonEmptySlice, NonEmptyVec};

fn main() {
    let vec = NonEmptyVec::<u8>::from_vec(vec![0]);
    let bad: &'static NonEmptySlice<_> = &vec;
    drop(vec);
    let _dangling = bad.first();
}
