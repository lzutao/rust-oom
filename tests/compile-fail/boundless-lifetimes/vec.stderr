error[E0597]: `vec` does not live long enough
 --> $DIR/vec.rs:5:42
  |
5 |     let bad: &'static NonEmptySlice<_> = &vec;
  |              -------------------------   ^^^^ borrowed value does not live long enough
  |              |
  |              type annotation requires that `vec` is borrowed for `'static`
...
8 | }
  | - `vec` dropped here while still borrowed

error[E0505]: cannot move out of `vec` because it is borrowed
 --> $DIR/vec.rs:6:10
  |
5 |     let bad: &'static NonEmptySlice<_> = &vec;
  |              -------------------------   ---- borrow of `vec` occurs here
  |              |
  |              type annotation requires that `vec` is borrowed for `'static`
6 |     drop(vec);
  |          ^^^ move out of `vec` occurs here
